llvm/docs/AMDGPUUsage.rst

```diff
@@ -5813,6 +5813,18 @@ in table :ref:`amdgpu-amdhsa-memory-model-code-sequences-gfx6-gfx9-table`.
                                                               be reordered by
                                                               hardware.
 
+     load         *none*       *none*         - constant - !volatile & !nontemporal
```

Comment:
Maybe this should say "constant address space **and** only if the compiler can prove that the address is uniform", but I'm not sure where to fit that into the table.

Should I mention scalar stores in the table? The text above says "Since the constant address space contents do not change during the execution of a kernel dispatch it is not legal to perform
stores".

Reply:
AFAIK we do not generate scalar stores as the instruction was removed from the isa.

I do not think you need to say "only is the address is uniform" as the compiler cannot generate scalar loads unless that condition is already met.